[INFO -- 2018-12-01 07:12:41]
Starting flip_value script
version: 1.1
[INFO -- 2018-12-01 07:12:41]
Initialization strings:
[INFO -- 2018-12-01 07:12:41]
(1) file /tmp/knapsack/ga
[INFO -- 2018-12-01 07:12:41]
initSignal:0
[INFO -- 2018-12-01 07:12:41]
endSignal:13
[INFO -- 2018-12-01 07:12:41]
faultModel:1
[EXCEPTION -- 2018-12-01 07:12:42]
pythonException: Unable to exit loop in struct fields; Exiting wihtout making a bit flip
[EXCEPTION -- 2018-12-01 07:12:42]
gdbException: Cannot access memory at address 0x555555f97bf0
[EXCEPTION -- 2018-12-01 07:12:42]
gdbException: Cannot access memory at address 0x557d55780520
[INFO -- 2018-12-01 07:12:42]
Backtrace BEGIN:
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:51
#1  0x00007ffff7490f5d in __GI_abort () at abort.c:90
#2  0x00007ffff7ae1095 in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff7adec86 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#4  0x00007ffff7adecd1 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#5  0x00007ffff7adef14 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00007ffff7b086f2 in std::__throw_bad_alloc() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#7  0x000055555555a2ee in __gnu_cxx::new_allocator<int>::allocate (this=0x7fffffffd900, __n=18446744030759878856) at /usr/include/c++/7/ext/new_allocator.h:102
#8  0x0000555555559815 in std::allocator_traits<std::allocator<int> >::allocate (__a=..., __n=18446744030759878856) at /usr/include/c++/7/bits/alloc_traits.h:436
#9  0x00005555555587e2 in std::_Vector_base<int, std::allocator<int> >::_M_allocate (this=0x7fffffffd900, __n=18446744030759878856) at /usr/include/c++/7/bits/stl_vector.h:172
#10 0x00005555555573be in std::vector<int, std::allocator<int> >::_M_allocate_and_copy<__gnu_cxx::__normal_iterator<int const*, std::vector<int, std::allocator<int> > > > (this=0x7fffffffd900, __n=18446744030759878856, __first=<error reading variable: Cannot access memory at address 0x557d55780520>, __last=816) at /usr/include/c++/7/bits/stl_vector.h:1260
#11 0x000055555555640b in std::vector<int, std::allocator<int> >::operator= (this=0x7fffffffd900, __x=std::vector of length -42949672760, capacity -42949672760 = {...}) at /usr/include/c++/7/bits/vector.tcc:206
#12 0x00005555555552ad in crossover (pai=std::vector of length 200, capacity 200 = {...}, mae=std::vector of length -42949672760, capacity -42949672760 = {...}, filho1=std::vector of length 200, capacity 200 = {...}, filho2=std::vector of length 0, capacity 0) at knapsackga.cpp:77
#13 0x00005555555555c7 in genetic_algorithm () at knapsackga.cpp:104
#14 0x0000555555555a9b in main () at knapsackga.cpp:141

46	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

Backtrace END
Memory content before bitflip:0x280x000x000x00
Fault Model: Double bit-flip
base address to flip value: 0x55555575f010 
address max offset: 4
Memory content after  bitflip:0x0a0x000x000x00
frame name: std::_Vector_base<int, std::allocator<int> >::_M_allocate
symbol name: pop_size
symbol filename: knapsackga.cpp
symbol line: 17
value: 10
value address: 0x55555575f010 <pop_size>
Type: The type is an integer type.
Type sizeof: 4

[INFO -- 2018-12-01 07:12:42]
Fault Injection Successful after 1.156684398651123s
[INFO -- 2018-12-01 07:12:42]
event type: exit
[EXCEPTION -- 2018-12-01 07:12:42]
exit code: no exit code available
